subprojects/core/src/main/java/org/gradle/initialization/RecordingClassLoaderScopeRegistry.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.initialization;
+
+import org.gradle.api.internal.initialization.ClassLoaderScope;
+import org.gradle.internal.classpath.ClassPath;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class RecordingClassLoaderScopeRegistry implements ClassLoaderScopeRegistry {
```

Comment:
I'd perhaps instead have a listener that receives notifications of scopes being created, and instant execution can listen for these events and collect whatever details it likes about the scopes.

Reply:
Done. I'm not so sure about the way I wired the listeners though.